File: src/l1mergeintervals/JS/mergeIntervals.js
function mergeIntervals(intervals) {
    if (intervals.length === 0) {
        return [];
    }

    const result = [];
    result.push(intervals[0]);

    for (let i = 1; i < intervals.length; i++) {
        const lastAddedInterval = result[result.length - 1];
        const curStart = intervals[i][0];
        const curEnd = intervals[i][1];

        if (lastAddedInterval[1] >= curStart) {
            lastAddedInterval[1] = Math.max(lastAddedInterval[1], curEnd);
        } else {
            result.push(intervals[i]);
        }
    }

    return result;
}

module.exports = { mergeIntervals };

File: src/l1mergeintervals/Python/merge_intervals.py
from typing import List

def merge_intervals(intervals: List[List[int]]) -> List[List[int]]:
    if not intervals:
        return []

    result = [intervals[0]]

    for i in range(1, len(intervals)):
        last_added_interval = result[-1]
        cur_start, cur_end = intervals[i]

        if last_added_interval[1] >= cur_start:
            last_added_interval[1] = max(last_added_interval[1], cur_end)
        else:
            result.append(intervals[i])

    return result
